File: backend/Entities/Vehicle.cs
namespace VehicleMaintenanceTracker.Entities;

public enum VehicleType
{
    Car,
    Truck,
    SnowPlow,
    ServiceVehicle
}

public class Vehicle
{
    public int Id { get; set; }
    public required string PlateNumber { get; set; }
    public required string Make { get; set; }
    public required string Model { get; set; }
    public int Year { get; set; }
    public int Mileage { get; set; }
    public VehicleType Type { get; set; }

    public List<MaintenanceRecord> MaintenanceRecords { get; set; } = [];
}

File: frontend/src/types/vehicle.ts
export type VehicleType = "Car" | "Truck" | "SnowPlow" | "ServiceVehicle";

export interface Vehicle {
  id: number;
  plateNumber: string;
  make: string;
  model: string;
  year: number;
  mileage: number;
  type: VehicleType;
  maintenanceCount: number;
}

export interface CreateVehicleRequest {
  plateNumber: string;
  make: string;
  model: string;
  year: number;
  mileage: number;
  type: VehicleType;
}

export const VEHICLE_TYPES: VehicleType[] = ["Car", "Truck", "SnowPlow", "ServiceVehicle"];

export const VEHICLE_TYPE_LABEL: Record<VehicleType, string> = {
  Car: "Car",
  Truck: "Truck",
  SnowPlow: "Snow plow",
  ServiceVehicle: "Service vehicle",
};
